modules/core/src/v2/keychains.ts

```diff
@@ -232,9 +232,6 @@ export class Keychains {
       }
     }
 
-    if (params.reqId) {
```

Comment:
This should stay in. I think the reason you were getting the CORS errors is because you were using the default environment (test) instead of the localNonSecure environment, which we recommend for doing local testing (although, I'm not sure this is an option for you as you aren't able to run our platform via localhost, but there is probably a workaround for this).

We should be allowing the Request-ID header to be used cross-domain though, so I'll look into updating the CORS allowed headers.

Reply:
Thanks for the detailed response, do let us know if there is a workaround. It looks to be returned in the Access-Control-Allow-Headers from other POST requests on the API when I had a look, so makes sense to have it here too.

Reply:
> Thanks for the detailed response, do let us know if there is a workaround. It looks to be returned in the Access-Control-Allow-Headers from other POST requests on the API when I had a look, so makes sense to have it here too.

Workaround for localhost testing would be to use socat or something similar to proxy requests. Here's a quick example:

In your terminal, run this command:
```
$ socat TCP-LISTEN:12345,fork,reuseaddr TCP:test.bitgo.com:443
```

This will make a port forward such that it appears that test.bitgo.com is available at localhost:12345 (you might need to change the port).

Then, in another terminal, you can test your connection by running this curl command:
```
$ curl -v --insecure https://localhost:12345/api/v2/ping
```

Note that --insecure is required, since we can't validate the certificate against localhost. Then you should be able to do local testing without running into CORS issues.

Reply:
That's a great point - noted!